cartographer/mapping_3d/sparse_pose_graph/optimization_problem.cc

```diff
@@ -154,6 +156,12 @@ void OptimizationProblem::Solve(const std::vector<Constraint>& constraints) {
             translation_parameterization(),
             common::make_unique<ceres::QuaternionParameterization>(), &problem);
       }
+      if (frozen) {
```

Comment:
Yes, this mirrored 2D. Since we currently add no nodes for maps this makes no difference in practice. I fixed in 2D and 3D anyway since we want to add nodes for maps in the future. PTAL